### PR TITLE
feat: Add support for ignoring experemental features

### LINKED ISSUE #199: Add the option to allow experimental features in `n/no-unsupported-features/node-builtins`
I am already loving the new builtin warnings, but I have realised that I want to allow all experimental features :eyes:

image

We do have the ignores option [no-unsupported-features/node-builtins.md#ignores](
https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/no-unsupported-features/node-builtins.md#ignores), but I think it'd be great to add an `allowExperimental` option.

### PR DESCRIPTION
**Pull request by scagood**
closes #199

### CODE REVIEW

File: lib/util/check-unsupported-builtins.js [hunk 1]
```diff
@@ -92,11 +97,14 @@ module.exports.checkUnsupportedBuiltins = function checkUnsupportedBuiltins(
 
     for (const { node, path, info } of references) {
         const name = unprefixNodeColon(path.join("."))
-        const supported = isSupported(info, options.version)
-
-        if (supported === true || options.ignores.has(name)) {
+        if (
+            isInRange(info.supported, options.version) ||
+            options.ignores.has(name) ||
+            (options.allowExperimental === true && info.experimental != null)
```
Comment (scagood) [inline lib/util/check-unsupported-builtins.js:103]: I wonder if we should check the current range against the experimental range here so we don't allow experimental features that are not in the configured node version

**Review by aladdin-add — CHANGES_REQUESTED** [review]
can you update the rule docs, otherwise LGTM, thanks!

File: docs/rules/no-unsupported-features/node-builtins.md [hunk 2]
```diff
@@ -36,6 +36,12 @@ But, you can overwrite the version by `version` option.
 
 The `version` option accepts [the valid version range of `node-semver`](https://github.com/npm/node-semver#range-grammar).
 
+#### allowExperimental
+
+This allows you to enable experimental features that are available in your configured node version
+
+The `"allowExperimental"` option accepts a boolean value
```
Comment (aladdin-add) [inline docs/rules/no-unsupported-features/node-builtins.md:43]:
```suggestion
The `"allowExperimental"` option accepts a boolean value (the default value is `false`).
```

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Pull request merged**